subsys/fs/littlefs_fs.c

```diff
@@ -509,6 +472,86 @@ static int littlefs_statvfs(struct fs_mount_t *mountp,
 	return lfs_to_errno(ret);
 }
 
+int fs_littlefs_read(const struct lfs_config *c, lfs_block_t block,
+			lfs_off_t off, void *buffer, lfs_size_t size)
+{
+	struct fs_littlefs_context *context = c->context;
+	struct lfs_config config_copy;
+
+	memcpy(&config_copy, c, sizeof(config_copy));
```

Comment:
Why is this copy needed (and also in the next routines)? What happens when someone reorganizes `lfs_config`?

Reply:
The context is used an identifier by the underlying device operations, like the flash area id for flash. Therefore, the device operations are able to set the context to a certain value in the call to fs->lfs_context.device_services.open. We have to use the context now as well, as we have to remember which device operations were used for this specific device. For this purpose I set the context of lfs_config as well, where the actual context of device_services.open is stored as well. For every hand over down to the device operation it is therefore necessary to restore the original context.

Reply:
A reorganization of lfs_config should only be problematic if context gets removed or renamed, but in such a case the build would fail anyway as this specific member is accessed by name.

Reply:
I have to reuse it as a consequence of https://github.com/zephyrproject-rtos/zephyr/pull/38029#discussion_r700828352. Otherwise I do not know in which context the LFS-operations are called (from LFS). The context allows me to figure out which device operations should be used for this specific LFS-operation.